tests/integration/local/secrets/store_test.go

```diff
@@ -139,6 +139,15 @@ func TestCreate_Secret(t *testing.T) {
 		require.Equal(t, "testValue", resolved)
 	})
 
+	t.Run("secret key should be all lowercase", func(t *testing.T) {
```

Comment:
Since the k8s secret naming convention is much more specific than just lower case letters (it also disallows certain symbols and has a length restriction), can you add a few more tescases and handle those as well?

Reply:
I updated the code to handle those